cg/meta/compress/compress.py

```diff
@@ -33,6 +40,17 @@ def set_dry_run(self, dry_run: bool):
         if self.crunchy_api.dry_run is False:
             self.crunchy_api.set_dry_run(dry_run)
 
+    def get_flow_cell_name(self, fastq_path: Path) -> str:
+        """
+        Extract the flow cell name from a fastq Path assuming fastq files are kept in their
+        demultipelxed path and the following run_name convention:
+
+            - <date>_<machine>_<run_numbers>_<A|B><flowcell_id>:
+            - Ex: 220128_A00689_0460_BHVN2FDSX2
+        """
+        run_name: str = fastq_path.relative_to(self.demux_root).parts[0]
```

Comment:
The issue is that a sample can be related to multiple flow cells. The flow cell, which is usually kept as a tag in housekeeper but since housekeeper loose the tag after decompression then we need a solve in another way. It is impossible to know if Illumina decides to change their naming structure, but it has remained consistent for a long time. We rely on the naming in cgstats and in demultiplexing procedures so IF it were to change then multiple stages in our business logic would fail. Sounds scary - yes - and could perhaps be something we should refactor into a regex constant which could be changed if this ever was to change?